docs/rfd.md

```diff
@@ -0,0 +1,478 @@
+
+   - cpu limits using cpu.max only, period hard coded to the default 100000, no weight/shares support, no cpuset support
+   - memory limits using memory.high and memory.max only, no swap or hugepages support
+   - device io limits using io.max only, no weight or latency support
+
+## Disclaimer
+
+This project is **not** intended for real-world use, and does not implement many of the the features that would be necessary to make it safe to use in a production environment.
+
+## Details
+
+The three main components of the job server are:
+
+1. A server that provides a GRPC API to allow clients to start and stop jobs, list and query the status of existing jobs, and stream logs of running or completed jobs.
+2. A set of generic and reusable library packages that are used to implement the server and API.
+3. A user-friendly CLI client that allows users to interact with the job server.
+
+### Proto Specification
+
+The Job Server API is composed of a single service with five methods, defined as follows:
+
+```protobuf
+syntax = "proto3";
+
+package job.v1;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/kralicky/jobserver/pkg/apis/job/v1";
+
+service Job {
+  // Starts a new job, and returns its id if it was started successfully.
+  //
+  // The job will be run asynchronously; this method does not wait for it
+  // to complete.
+  //
+  // If resource limits are provided, the job will be run in its own cgroup
+  // with all the given limits applied. Any combination of the allowed limit
+  // types may be provided; it is not necessary to set all of them.
+  rpc Start(JobSpec) returns (JobId);
+
+  // Stops a running job. Once stopped, this method will wait until the job has
+  // completed before returning.
+  //
+  // This will first attempt to stop the job's process using SIGTERM, but if the
+  // process does not exit within a short grace period, it will be forcefully
+  // killed with SIGKILL.
+  //
+  // A job must be in the Running state to be stopped. If the job is in any
+  // other state, this returns a FailedPrecondition error.
+  rpc Stop(JobId) returns (google.protobuf.Empty);
+
+  // Returns the status of an existing job.
+  //
+  // If the job is completed, the exit code and detailed wait status will be
+  // present in the response. Jobs stopped by the user with the Stop() method
+  // may not have an exit code, depending if the process was able to exit
+  // cleanly within the grace period. Jobs that failed to start, or were
+  // otherwise killed by the system, will not have an exit code, but may have
+  // a detailed wait status indicating the reason for failure.
+  rpc Status(JobId) returns (JobStatus);
+
+  // Returns a list of all job IDs that are currently known to the server.
+  //
+  // All jobs, regardless of state, are included. No guarantees are made about
+  // the order of ids in the list.
+  rpc List(google.protobuf.Empty) returns (JobIdList);
```

Comment:
This is not in the spec of the challenge - it's easy to implement from an API perspective but getting a table printed out _well_ in a terminal is fraught with peril. I'd drop this 😅

Reply:
Haha, I went back and forth about whether or not to include this, it was just slightly ambiguous enough in the spec where I thought it could have been deliberately left out as an 'exercise to the reader' so to speak :laughing: 

For printing tables, I am a fan of `"github.com/jedib0t/go-pretty/v6/table"`, the implementation would look like:

```go
list, err := client.List(ctx, &emptypb.Empty{})
if err != nil {
	return err
}
tab := table.NewWriter()
tab.AppendHeader(table.Row{"JOB ID", "COMMAND", "CREATED", "STATUS"})
for _, id := range list.Items {
	stat, err := client.Status(cmd.Context(), id)
    // handle err
	tab.AppendRow(table.Row{
		id.GetId(),
		stat.GetSpec().GetCommand().GetCommand(),
		stat.GetStartTime().AsTime(),
		stat.GetMessage(),
	})
}
fmt.Println(tab.Render())
```

I can omit List if you want, but I do think it's rather important to the overall UX.

Reply:
It's your call, it's not required for the challenge but it will be reviewed if it's in there. 😇